dhall/src/semantics/resolve/resolve.rs

```diff
@@ -303,6 +322,51 @@ pub(crate) fn download_http_text(_url: Url) -> Result<String, Error> {
     panic!("Remote imports are not supported on wasm yet")
 }
 
+#[cfg(all(not(target_arch = "wasm32"), feature = "reqwest"))]
+pub(crate) fn cors_check(
+    origin: Option<String>,
+    remote: &Url,
+) -> Result<(), Error> {
+    if origin.is_none() {
+        return Ok(());
+    }
+    let origin = origin.unwrap();
+    let remote_origin =
+        format!("{}://{}", remote.scheme(), remote.host_str().unwrap());
```

Comment:
I think you should use https://docs.rs/reqwest/0.11.3/reqwest/struct.Url.html#method.origin here to be properly standard-compliant